z80count/z80count.py

```diff
@@ -35,49 +35,138 @@
 def z80count(line,
              parser,
              total,
-             total_cond,
              subt,
              no_update,
-             tabstop=2,
+             column=50,
              debug=False,
+             use_tabs=False,
+             tab_width=8,
              ):
     out = line.rstrip() + "\n"
     entry = parser.lookup(line)
     if entry:
-        cycles = entry["cycles"]
-        if "/" in cycles:
-            c = cycles.split("/")
-            total_cond = total + int(c[0])
-            total = total + int(c[1])
+        total, total_cond = update_counters(entry, total)
+        out = format_line(
+            line, entry, total, total_cond, subt, update=not no_update,
+            column=column, debug=debug, use_tabs=use_tabs,
+            tab_width=tab_width,
+        )
+    return (out, total)
+
+
+def update_counters(entry, total):
+    if entry["_t_states_met"]:
+        total_cond = total + entry["_t_states_met"]
+    else:
+        total_cond = 0
+    total = total + entry["_t_states_or_not_met"]
+
+    return (total, total_cond)
+
+
+def format_line(line, entry, total, total_cond, subt, update, column,
+                debug, use_tabs, tab_width):
+    cycles = entry["cycles"]
+    line = line.rstrip().rsplit(";", 1)
+    comment = "; [%s" % cycles
+    if subt:
+        if total_cond:
+            comment += " .. %d/%d]" % (total_cond, total)
         else:
-            total = total + int(cycles)
-            total_cond = 0
-
-        line = line.rstrip().rsplit(";", 1)
-        comment = "; [%s" % cycles
-        if subt:
-            if total_cond:
-                comment += " .. %d/%d]" % (total_cond, total)
-            else:
-                comment += " .. %d]" % total
+            comment += " .. %d]" % total
+    else:
+        comment += "]"
+    if debug:
+        comment += " case{%s}" % entry["case"]
+
+    if len(line) == 1:
+        comment = comment_alignment(line[0], column, use_tabs, tab_width) + comment
+    out = line[0] + comment
+    if len(line) > 1:
+        if update:
+            m = OUR_COMMENT.search(line[1])
+            if m:
+                line[1] = line[1].replace(m.group(0), "")
+        out += " "
+        out += line[1].lstrip()
+    out += "\n"
+
+    return out
+
+
+def comment_alignment(line, column, use_tabs=False, tab_width=8):
+    """Calculate the spacing required for comment alignment.
+
+    :param str line: code line
+    :param int column: column in which we want the comment to start
+    :param bool use_tabs: use tabs instead of spaces
+    :param int tab_width: tab width
+
+    :returns: the spacing
+    :rtype: str
+
+    """
+
+    # The code is a bit obscure. I hope this diagram helps to clarify.
```

Comment:
I rather have comments inline instead of ascii art that is hard to maintain and will be outdated essentially.

(If actually needed because reading the code is not that difficult to understand)

Reply:
done